### PR TITLE
Add url properties for Spreadsheet and Worksheet

### PR DESCRIPTION
**Pull request by CrossNox**
Hi! I didn't find any contribution guidelines so please tell me how I can improve this PR :smile: 

# Why
Could be useful to have the drive URL to share. Our workflows use `gspread-pandas` to build reports and share them. Their `Spread` class has an `url` property which is very useful, but IMHO it should be here. Also, we need the URL for individual worksheets.

# What
This PR adds a `url` property for both `Spreadsheet` and `Worksheet` classes.

### CODE REVIEW

**Comment by burnash** [comment]
Hi there, 
Thank you for your contribution.

I try to keep the API as minimal as possible. I need to check if this addition fits this idea.

Regarding the contribution guidelines, here they are: https://github.com/burnash/gspread/blob/master/.github/CONTRIBUTING.md

They are linked from the README, but I guess they aren't prominent enough.
https://github.com/burnash/gspread#contribute-code

Where are you usually find them? (I've been away from GitHub lately and maybe I miss some new feature where the GitHub links the guidelines somewhere automatically)

**Comment by CrossNox** [comment]
Sorry, my bad! I missed those. I usually find them at the root of the project, but github clearly finds them there, so that's on me. 

Would you rather stall this PR and discuss this on an issue as the contributing guidelines state?

**Comment by burnash** [comment]
No problem :)
And no need to stall the PR. Let's add these properties.

This PR is missing tests though. Could you please add them? Travis is failing for some reason, however, it should work locally. 

Here's how to set up testing:
https://github.com/burnash/gspread/blob/master/.github/CONTRIBUTING.md#testing

File: gspread/urls.py [hunk 1]
```diff
@@ -19,3 +19,6 @@
 DRIVE_FILES_API_V2_URL = 'https://www.googleapis.com/drive/v2/files'
 DRIVE_FILES_UPLOAD_API_V2_URL = ('https://www.googleapis.com'
                                  '/upload/drive/v2/files')
+
+SPREADSHEET_DRIVE_URL = "https://docs.google.com/spreadsheets/d/%s"
+WORKSHEET_DRIVE_URL = SPREADSHEET_URL + "#gid=%s"
```
Comment (burnash) [inline gspread/urls.py]: I noticed that you're using a different template for `WORKSHEET_DRIVE_URL` than in the Sheets UI. Where it's:
```
https://docs.google.com/spreadsheets/d/spreadsheetId/edit#gid=sheetId
```
As described in https://developers.google.com/sheets/api/guides/concepts

What was the reason for dropping the `/edit` part? Looks like the Google server issues a 302 redirect when I try to access the spreadsheet without `/edit` part:

```
$ curl -I https://docs.google.com/spreadsheets/d/my_SeCr4tSh44t-111/#gid=0
HTTP/2 302
content-type: text/html; charset=UTF-8
cache-control: no-cache, no-store, max-age=0, must-revalidate
pragma: no-cache
expires: Mon, 01 Jan 1990 00:00:00 GMT
date: Tue, 18 Feb 2020 11:48:45 GMT
location: https://docs.google.com/spreadsheets/d/my_SeCr4tSh44t-111/edit
```

**Comment by burnash** [comment]
I guess tests were too much to ask for :) I decided it'd be easier to merge it now.
Thank you again for your contribution!

**Comment by CrossNox** [comment]
Sorry! my laptop died and haven't had much time to follow up on this (finals, quarantine, etc). The tests upstream were failing for me and I was digging into why last time I had time for it. Guess I'll open a new PR when I have the time for it.

**Comment by burnash** [comment]
No worries. I don't see any rudeness in the comment though: I also lack the time and that's why I added the guidelines in the first place. I understand that writing tests is tough for this type of project since the external API is involved. But they make the maintainer's life much easier.

**Pull request merged**